### PR TITLE
Add recommendation to enable 'preserve log'

### PR DESCRIPTION
**Pull request by EdmondChuiHW**
# Summary

When the web socket between CDT and dev-middleware is closed, CDT consults the "Preserve log" setting and clears log upon disconnection accordingly.

This can be unexpected for React Native targets, as users may expect to continue inspecting the console messages even if the native app has crashed.

In this PR, we print a recommendation for Fusebox targets with a button to direct the user to the specific setting.

Related: UI improvement for users viewing console messages under the disconnection state: https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/116

# Test plan

Conenct to a Fusebox target. The recommendation should show up if the "Preserve log" setting is off:
screenshot of the new console message
Notice the yellow highlight, but it's not a warning (does not increase the warning counter in the top-right).

Clicking on the "show settings" link opens the Settings panel and highlights the checkbox with an animation
freeze frame of the animation highlighting the settings entry

Video:

https://github.com/user-attachments/assets/3fc6cce7-c137-4763-a39f-9d953223146a

# Upstreaming plan

- [ ] This commit should be sent as a patch to the upstream `devtools-frontend` repo. I've reviewed the [contribution guide](https://docs.google.com/document/d/1WNF-KqRSzPLUUfZqQG5AFeU_Ll8TfWYcJasa_XGf7ro/edit#heading=h.9kj7femz1xg5).
- [x] This commit is React Native-specific and cannot be upstreamed.

### CODE REVIEW

File: front_end/entrypoints/rn_fusebox/rn_fusebox.ts [hunk 1]
```diff
@@ -106,6 +106,18 @@ class FuseboxClientMetadataModel extends SDK.SDKModel.SDKModel<void> {
           const success = false;
           Host.rnPerfMetrics.fuseboxSetClientMetadataFinished(success, reason);
         });
+
+    if (!Common.Settings.Settings.instance().moduleSetting('preserve-console-log').get()) {
+      target.model(SDK.ConsoleModel.ConsoleModel)
+          ?.addMessage(new SDK.ConsoleModel.ConsoleMessage(
+              target.model(SDK.RuntimeModel.RuntimeModel), Protocol.Log.LogEntrySource.Recommendation,
+              Protocol.Log.LogEntryLevel.Info,
+              '[React Native] Console messages are currently cleared upon DevTools disconnection. You can preserve logs in settings: ',
```
Comment (EdmondChuiHW) [inline front_end/entrypoints/rn_fusebox/rn_fusebox.ts:115]: Text suggestions welcome!

File: front_end/ui/components/settings/settingCheckbox.css [hunk 2]
```diff
@@ -5,6 +5,7 @@
  */
 
 :host {
+  display: block;
```
Comment (EdmondChuiHW) [inline front_end/ui/components/settings/settingCheckbox.css:8]: Needed for the existing animation util to work.

The util attempts to animation the background on a Shadow DOM element `<setting-checkbox>`:

https://github.com/facebookexperimental/rn-chrome-devtools-frontend/blob/8de9257f1ca9f129a763165e6df7dac9583f3eb7/front_end/panels/utils/utils.ts#L110

But since the host element is not a `block` by default, the background wouldn't be applied:
https://stackoverflow.com/a/60702911

File: front_end/entrypoints/rn_fusebox/rn_fusebox.ts (hunk 1 above)
Comment (vzaidman) [reply front_end/entrypoints/rn_fusebox/rn_fusebox.ts:115]: Do we support other languages besides English?

**Comment by vzaidman** [comment]
Could we **also** add that same message **after** the devices disconnects and logs are cleared?

**Comment by huntie** [comment]
@vzaidman Ditto. Can we _**exclusively**_ log this after the device disconnects?

**Comment by EdmondChuiHW** [comment]
@vzaidman @huntie this has been explored. It's possible to impl, but not ergonomic to do it in a Fusebox-only/merge-conflict-minimal way, i.e. via the regular event listeners. 

TLDR: will be ugly; can do if there's a strong preference

**Comment by vzaidman** [comment]
Displaying it after the disconnect is the important one despite the ugliness.

I do think there's value in displaying it before, but we might as well skip it to not introduce extra deviations from the main branch.

**Comment by EdmondChuiHW** [comment]
Let's do it. Besides the code, a heads up on the UI: the message will be covered up by the "disconnected" dialog.    That's what made me add it to the dialog initially, but I agree adding more to the dialog could get bloated.

**Comment by EdmondChuiHW** [comment]
New UI:


screenshot of the prompt in a console message behind the 'disconnected' dialog

**Comment by vzaidman** [comment]
Nice! Can you actually click on "Show Settings"?

**Comment by EdmondChuiHW** [comment]
> Nice! Can you actually click on "Show Settings"?

@vzaidman Yup! With an animation to highlight the checkbox. Check out the video in the original description

**Comment by vzaidman** [comment]
I'm asking because the popup seems to be hiding the logs at the moment. Can you close it?

**Comment by EdmondChuiHW** [comment]
Yup. The "Dismiss" button in the dialog works the same way (doesn't clear this new message)

**Review by vzaidman — APPROVED** [review]

File: front_end/entrypoints/rn_fusebox/rn_fusebox.ts (hunk 1 above)
Comment (EdmondChuiHW) [reply front_end/entrypoints/rn_fusebox/rn_fusebox.ts]: Sorry missed this. Yes I should use an i18n string. Don't think we're currently doing any translation at the moment but good practice regardless. Will follow up
